AWSAppSyncClient/AWSAppSyncClient.swift

```diff
@@ -487,22 +484,26 @@ public class AWSAppSyncClientInfo {
     }
     
     public init(forKey: String) throws {
-        if AWSInfo.default().rootInfoDictionary["AppSync"] == nil {
-            throw AWSAppSyncClientInfoError(errorMessage: "Cannot read configuration from the awsconfiguration.json/Info.plist")
-        }
-        
-        let appSyncConfig : [String : Any] = (AWSInfo.default().rootInfoDictionary["AppSync"] as? [String : Any])!
-        let defaultAppSyncConfig : [String : Any] = (appSyncConfig[forKey] as? [String : Any])!
-        self.apiUrl = defaultAppSyncConfig["ApiUrl"] as! String
-        self.region = defaultAppSyncConfig["Region"] as! String
-        self.authType = defaultAppSyncConfig["AuthMode"] as! String
-        
-        if let apiKeyFromDictionary = defaultAppSyncConfig["ApiKey"] {
-            self.apiKey = apiKeyFromDictionary as! String
-        } else {
-            if (self.authType == AuthType.apiKey.rawValue) {
-                throw AWSAppSyncClientInfoError(errorMessage: "API_KEY AuthMode found in configuration but a valid ApiKey is not found")
+        do {
+            if AWSInfo.default().rootInfoDictionary["AppSync"] == nil {
+                throw AWSAppSyncClientInfoError(errorMessage: "Cannot read configuration from the awsconfiguration.json/Info.plist")
             }
+            
+            let appSyncConfig : [String : Any] = (AWSInfo.default().rootInfoDictionary["AppSync"] as? [String : Any])!
+            let defaultAppSyncConfig : [String : Any] = (appSyncConfig[forKey] as? [String : Any])!
+            self.apiUrl = defaultAppSyncConfig["ApiUrl"] as! String
+            self.region = defaultAppSyncConfig["Region"] as! String
+            self.authType = defaultAppSyncConfig["AuthMode"] as! String
+            
+            if let apiKeyFromDictionary = defaultAppSyncConfig["ApiKey"] {
+                self.apiKey = apiKeyFromDictionary as! String
+            } else {
+                if (self.authType == AuthType.apiKey.rawValue) {
+                    throw AWSAppSyncClientInfoError(errorMessage: "API_KEY AuthMode found in configuration but a valid ApiKey is not found")
+                }
+            }
+        } catch {
+            throw AWSAppSyncClientInfoError(errorMessage: "Error in reading AppSync configuration from the awsconfiguration.json/Info.plist")
```

Comment:
We should remove Info.plist from here (and other places where we have used it)

Reply:
Thank you for pointing it out. Removed references to Info.plist.